tests/unit/test_strike_service.py

```diff
@@ -0,0 +1,141 @@
+import pytest
+from pytest_mock.plugin import MockerFixture
+from moddingway.database.models import Strike
+from moddingway.services import strike_service
+from moddingway import enums
+from typing import List
+import discord
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "previous_points,total_points,expected_punishment",
+    [
+        (0, 1, "Nothing"),
+        (1, 8, "11 day exile"),
+        (8, 10, "14 day exile"),
+        (10, 11, "14 day exile"),
+        (1, 9, "11 day exile"),
+        (9, 12, "14 day exile"),
+        (12, 13, "14 day exile"),
+        (10, 15, "Permanent ban"),
+    ],
+)
+async def test_apply_punisment(
+    previous_points: int,
+    total_points: int,
+    expected_punishment: str,
+    mocker: MockerFixture,
+    create_db_user,
+):
+    mocked_logging_embed = mocker.Mock()
+    mocked_user = mocker.Mock(roles=[enums.Role.VERIFIED])
```

Comment:
It doesn't matter in this case because the code you have isn't looking at the user's roles, but since this is mocking a discord version of a user, you need to pass a mocked version of the `role` object into the roles array. We already have a fixture that builds the mocked role, and I'm putting in a fixture to build the mocked user (or technically a discord member) that lets us just pass in the enum directly.